pkg/utils/utils.go

```diff
@@ -918,23 +917,29 @@ func GetBackupdriverClient(config *rest.Config) (v1alpha1.BackupdriverV1alpha1In
 }
 
 // Provide a utility function in guest cluster to clean up corresponding supervisor cluster snapshot CR
-func DeleteSvcSnapshot(gcSnapshot *backupdriverapi.Snapshot, svcConfig *rest.Config, svcNamespace string, logger logrus.FieldLogger) error {
-	svcSnapshotName := gcSnapshot.Status.SvcSnapshotName
+func DeleteSvcSnapshot(svcSnapshotName string, gcSnapshotName string, gcSnapshotNamespace string, guestConfig *rest.Config, logger logrus.FieldLogger) error {
 	if svcSnapshotName == "" {
-		logger.Errorf("No corresponding supervisor cluster snapshot CR name for guest cluster snapshot %s", gcSnapshot.Name)
-		return errors.New("No corresponding supervisor cluster snapshot CR.")
+		errMsg := fmt.Sprintf("No corresponding supervisor cluster snapshot CR name for guest cluster snapshot %s/%s", gcSnapshotNamespace, gcSnapshotName)
+		logger.Error(errMsg)
+		return errors.New(errMsg)
+	}
+	svcConfig, svcNamespace, err := GetSupervisorConfig(guestConfig, logger)
+	if err != nil {
+		logger.WithError(err).Error("Failed to get supervisor config from given guest config")
+		return err
 	}
 	backupdriverClient, err := GetBackupdriverClient(svcConfig)
 	if err != nil {
-		logger.WithError(err).Errorf("Failed to retrieve plugin client from svcConfig %v", svcConfig)
+		logger.WithError(err).Error("Failed to retrieve plugin client from given svcConfig")
```

Comment:
svcConfig -> guestConfig

Reply:
This err is returned by GetBackupdriverClient which uses svcConfig in line 931.